presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -89,6 +89,11 @@ private Scope(
         return Optional.empty();
     }
 
+    public Optional<Scope> getParent()
+    {
+        return parent;
+    }
```

Comment:
When one should call `.getParent()` and when `. getOuterQueryParent()`?

Reply:
when we resolve a field, we recurse through the parents independent of whether it crosses a query boundary and stop when we find a match, so we need to do the same process to get the field that goes with the fieldId.  If you want specifically to get the inner or outer query you would call getInnerQueryParent() or getOuterQueryParent()